### ISSUE TITLE
Publications waiting on other shards to become ready

### ISSUE THREAD

**Issue by psFried** [estuary/animated-carnival#7]
What we observed in the UI is that, after submitting a build with `taskA`, that `taskB` also gets pulled into the build. `taskB` gets included because it reads from or writes to the same collection as `taskA`. It's unclear whether the draft includes the collection that both tasks depend on, and it's also not clear whether or not that should make a difference in terms of the desired behavior. In other words, it's possible that we should address this by ensuring that the collection doesn't get added to the draft if it's not being modified. I think our first step here needs to be to reproduce the issue and get a better understanding of how it's happening before we decide exactly what the desired behavior should be. But the desired _outcome_ is that users should not be surprised by the inclusion of another task.

**Comment by jgraettinger** [comment]
this was fixed by adding a --no-wait flag to `api activate`, which the agent uses.
